core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -1472,7 +1467,7 @@ class PlaintextConsumerTest extends BaseConsumerTest {
     val fetchLead = consumer.metrics.get(new MetricName("records-lead", "consumer-fetch-manager-metrics", "", tags))
     assertNotNull(fetchLead)
 
-    assertTrue(s"The lead should be ${records.count}", records.count == fetchLead.metricValue())
+    assertEquals(fetchLead.metricValue().asInstanceOf[Double], records.count.asInstanceOf[Double], s"The lead should be ${records.count}")
```

Comment:
`toDouble` should be used instead of `asInstanceOf[Double]` for numbers. `asInstanceOf` should only be used for casting. It seems that we don't even need to convert the second parameter though.